File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cs
public class Solution {
    public bool CheckDivisibility(int n) {
        int sum=0;
        int product=1;
        int x=n;

        // Har digit ko nikalenge
        while(x!=0){
            int digit=x%10;
            x/=10;
            sum+=digit;
            product*=digit;
        }
        // Divisibility check
        return n%(sum+product)==0;
    }
}

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self,n:int)->bool:
        sum=0
        product=1
        x=n

        # Har digit ko nikalenge
        while x!=0:
            digit=x%10
            x//=10
            sum+=digit
            product*=digit

        # Divisibility check
        return n%(sum+product)==0